### PR TITLE
[textmate] Added Textmate grammars for several languages 

### PR DESCRIPTION
**Pull request by svenefftinge**
Also removed monaco-languages, which forcefully overwrites grammars
for many languages. Replaced with a textmate equivalent.

### CODE REVIEW

File: packages/cpp/data/c.tmLanguage.json [hunk 1]
```diff
@@ -0,0 +1,1955 @@
+{
+	"information_for_contributors": [
+		"This file has been converted from https://github.com/atom/language-c/blob/master/grammars/c.cson",
+		"If you want to provide a fix or improvement, please create a pull request against the original repository.",
+		"Once accepted there, we are happy to receive an update request."
+	],
+	"version": "https://github.com/atom/language-c/commit/9c0c5f202741a5647025db8d5df5fefba47b036c",
```
Comment (AlexTugarev) [inline packages/cpp/data/c.tmLanguage.json]: do we require to add copyright info for the grammars?

File: packages/python/src/browser/python-frontend-module.ts [hunk 2]
```diff
@@ -17,8 +17,12 @@
 import { ContainerModule } from "inversify";
 import { LanguageClientContribution } from "@theia/languages/lib/browser";
 import { PythonClientContribution } from "./python-client-contribution";
+import { PythonGrammarContribution } from "./python-grammar-contribution";
+import { LanguageGrammarDefinitionContribution } from "@theia/monaco/lib/browser/textmate";
```
Comment (AlexTugarev) [inline packages/python/src/browser/python-frontend-module.ts]: dependency to `@theia/monaco` is missing in `package.json`

Comment (AlexTugarev) [reply packages/python/src/browser/python-frontend-module.ts]: `files.data` property is missing, too

**Comment by AlexTugarev** [comment]
screen shot 2018-07-03 at 12 38 13

Java code looks better now. Well, almost.

**Comment by AlexTugarev** [comment]
Sometimes, you can see the application of decorators.

**Comment by AlexTugarev** [comment]
screen shot 2018-07-03 at 13 03 54

yaml coloring looks incomplete, no?

**Comment by AlexTugarev** [comment]
screen shot 2018-07-03 at 13 15 50

shell scripts

**Comment by AlexTugarev** [comment]
screen shot 2018-07-03 at 13 24 20

python looks good!

**Comment by AlexTugarev** [comment]
screen shot 2018-07-03 at 13 26 19

html looks bogus to me. at least `<` and `>` are not right, no? compare with `</`

**Comment by AlexTugarev** [comment]
screen shot 2018-07-03 at 13 29 16

headings are not colored in md.

File: packages/cpp/data/c.tmLanguage.json (hunk 1 above)
Comment (svenefftinge) [reply packages/cpp/data/c.tmLanguage.json]: They have pointers to their original location (incl license). We do the same as VSCode does.

**Comment by svenefftinge** [comment]
The minor oddities in highlighting come from https://github.com/theia-ide/theia/blob/master/packages/monaco/src/browser/textmate/textmate-tokenizer.ts#L50

**Comment by svenefftinge** [comment]
> Sometimes, you can see the application of decorators.

Textmate is slower than monarch. I see the recolorization in VSCode, too.
That said the native library is faster, of course.

File: packages/python/src/browser/python-frontend-module.ts (hunk 2 above)
Comment (svenefftinge) [reply packages/python/src/browser/python-frontend-module.ts]: done

**Pull request merged**